core/templates/dev/head/pages/exploration_editor/translation_tab/StateTranslationEditorDirective.js

```diff
@@ -0,0 +1,149 @@
+              contentId)) {
+              $uibModal.open({
+                templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                  '/components/forms/' +
+                  'mark_all_audio_as_needing_update_modal_directive.html'),
+                backdrop: true,
+                resolve: {
+                  considerTranslation: function() {
+                    return false;
+                  }
+                },
+                controller: 'MarkAllAudioAsNeedingUpdateController'
+              }).result.then(function() {
+                contentIdsToAudioTranslations.markAllAudioAsNeedingUpdate(
+                  contentId);
+                ExplorationStatesService.saveContentIdsToAudioTranslations(
+                  stateName, contentIdsToAudioTranslations);
+              });
+            }
+          };
+          var contentId = null;
+          var langaugeCode = null;
+          $scope.activeWrittenTranslation = null;
+          $scope.translationEditorIsOpen = false;
+          $scope.isTranslatable = EditabilityService.isTranslatable;
+          $scope.$on('activeContentIdChanged', function() {
+            initEditor();
+          });
+          $scope.$on('activeLanguageChanged', function() {
+            initEditor();
+          });
+
+          var initEditor = function() {
+            $scope.activeWrittenTranslation = null;
+            $scope.translationEditorIsOpen = false;
+            contentId = (
+              TranslationTabActiveContentIdService.getActiveContentId());
+            langaugeCode = TranslationLanguageService.getActiveLanguageCode();
+            if (StateWrittenTranslationsService.displayed.hasWrittenTranslation(
+              contentId, langaugeCode)) {
+              $scope.activeWrittenTranslation = (
+                StateWrittenTranslationsService.displayed
+                  .getWrittenTranslation(contentId, langaugeCode));
+            }
+          };
+          initEditor();
+          var saveTranslation = function() {
+            var oldTranslation = '';
+            var newTranslation = '';
+            contentId = (
+              TranslationTabActiveContentIdService.getActiveContentId());
+            langaugeCode = TranslationLanguageService.getActiveLanguageCode();
+            if (StateWrittenTranslationsService
+              .savedMemento.hasWrittenTranslation(contentId, langaugeCode)) {
+              oldTranslation = (StateWrittenTranslationsService
+                .savedMemento.getWrittenTranslation(contentId, langaugeCode));
+            }
+            newTranslation = (StateWrittenTranslationsService
+              .displayed.getWrittenTranslation(contentId, langaugeCode));
+            if (oldTranslation !== newTranslation) {
+              var stateName = StateEditorService.getActiveStateName();
+              showMarkAllAudioAsNeedingUpdateModalIfRequired(contentId);
+              StateWrittenTranslationsService.saveDisplayedValue();
```

Comment:
You should probably do this after the data is actually stored (The next line of code). That way if the actual save fails for some reason, you wouldn't update the savedMemento.